lib/queue.js

```diff
@@ -1174,49 +1174,20 @@ Queue.prototype.clean = function(grace, type, limit) {
  * @returns {Promise}
  */
 Queue.prototype.whenCurrentJobsFinished = function() {
-  return new Promise((resolve, reject) => {
-    if (!this.bclientInitialized) {
-      // bclient not yet initialized, so no jobs to wait for
-      return resolve();
-    }
-
-    //
-    // Force reconnection of blocking connection to abort blocking redis call immediately.
-    //
-    const forcedReconnection = redisClientDisconnect(this.bclient).then(() => {
-      return this.bclient.connect();
-    });
+  if (!this.bclientInitialized) {
+    // bclient not yet initialized, so no jobs to wait for
+    return Promise.resolve();
+  }
 
-    Promise.all(this.processing)
-      .then(() => {
-        return forcedReconnection;
-      })
-      .then(resolve, reject);
-
-    /*
-    this.bclient.disconnect();
-    this.bclient.once('end', function(){
-      console.error('ENDED!');
-      setTimeout(function(){
-        this.bclient.connect();
-      }, 0);
-    });
+  //
+  // Force reconnection of blocking connection to abort blocking redis call immediately.
+  //
+  const forcedReconnection = redisClientDisconnect(this.bclient).then(() => {
+    return this.bclient.connect();
+  });
 
-    /*
-    var stream = this.bclient.connector.stream;
-    if(stream){
-      stream.on('finish', function(){
-        console.error('FINISHED!');
-        this.bclient.connect();
-      });
-      stream.on('error', function(err){
-        console.error('errir', err);
-        this.bclient.connect();
-      });
-      this.bclient.connect();
-    }
-    */
-    //this.bclient.connect();
+  return Promise.all([this.processing[0]]).then(() => {
```

Comment:
After upgrade to latest version of bull we are seeing that on shutdown, it doesn't properly wait for all jobs to complete. 

It seems to me that this is likely the change that caused the problem. Was it intentional to change this to only wait for the first job?

Reply:
No, that seems like a dumb mistake on my part. Thanks for catching it @holm.

Reply:
Fix: https://github.com/OptimalBits/bull/pull/1586

Reply:
Thanks for the quick fix!